src/LtiMessageLaunch.php

```diff
@@ -118,52 +120,27 @@ public static function fromCache(
         ICache $cache = null,
         ILtiServiceConnector $serviceConnector = null
     ) {
+        // @TODO: Fix the null here on the next major version
         $new = new LtiMessageLaunch($database, $cache, null, $serviceConnector);
         $new->launch_id = $launch_id;
         $new->jwt = ['body' => $new->cache->getLaunchData($launch_id)];
 
         return $new->validateRegistration();
     }
 
-    public function setRequest(array $request = null)
+    public function setRequest(array $request)
     {
-        if ($request === null) {
-            $request = $_POST;
-        }
-
         $this->request = $request;
 
         return $this;
     }
 
-    private function shouldMigrate(): bool
-    {
-        assert($this->db instanceof IMigrationDatabase);
-
-        $launchData = $this->getLaunchData();
-        $lti1p1Install = $this->db->getMatchingLti1p1Install($launchData);
-        if ($lti1p1Install !== null) {
-            return $this->lti1p1InstallationSignatureMatches($lti1p1Install, $launchData);
-        }
-
-        return false;
-    }
-
-    private function lti1p1InstallationSignatureMatches(Lti1p1Installation $ltiInstall, array $launchData): bool
+    public function initialize(array $request)
     {
-        // Check to see if we have params to calculate oauth_consumer_key_sign
-        if (!isset($launchData[LtiConstants::LTI1P1]) || !isset($launchData[LtiConstants::LTI1P1]['oauth_consumer_key_sign'])) {
-            throw new LtiException(static::ERR_MISSING_OAUTH_CONSUMER_KEY_SIGN);
```

Comment:
The reason I distinguished between not having the key_sign passed, and the key_sign not matching, was so we could give different messages to a user who is attempting this launch, as there will be two different things needed to be done in order to fix the issue. if they simply don't have the key_sign, then the deployment needs to verified manually. And they should not be able to proceed as the deployment was missing. But if we only rely on the "deployment is missing" error, I can't know that it was an attempted migration. The other error is more significant in that it had the proper things to migrate automatically, but the key sign didn't match. meaning the LMS is doing something wrong with their 1.3 launch and something needs to be rectified between Platform and Tool.

Reply:
So then, should the `if` statement here actually be this? ...
```php
if (isset($launchData[LtiConstants::LTI1P1]) && !isset($launchData[LtiConstants::LTI1P1]['oauth_consumer_key_sign']))
```

That is, we have data for an LTI 1.1 migration, but they didn't send over the sign key

Reply:
I've got some open questions related to this with IMS Global, but I'll make sure different error messages are included in the final PR. I'm gonna merge this for now, tho.